nx/lib/nx/defn/grad.ex

```diff
@@ -406,13 +406,18 @@ defmodule Nx.Defn.Grad do
     to_grad(x, g, cache)
   end
 
-  defp grad(:sqrt, [x], ans, g, cache) do
-    g = Nx.multiply(g, Nx.divide(0.5, ans))
+  defp grad(:arcsin, [x], _ans, g, cache) do
+    g = Nx.divide(g, Nx.power(Nx.subtract(1.0, Nx.power(x, 2.0)), 0.5))
     to_grad(x, g, cache)
   end
 
-  defp grad(:tanh, [x], ans, g, cache) do
-    g = Nx.multiply(g, Nx.subtract(1.0, Nx.multiply(ans, ans)))
+  defp grad(:cos, [x], _ans, g, cache) do
+    g = Nx.multiply(g, Nx.negate(Nx.sin(x)))
+    to_grad(x, g, cache)
+  end
+
+  defp grad(:arccos, [x], _ans, g, cache) do
+    g = Nx.divide(g, Nx.multiply(-1.0, Nx.power(Nx.subtract(1.0, Nx.power(x, 2.0)), 0.5)))
```

Comment:
This is me just nitpicking:

```suggestion
    g = Nx.multiply(g, Nx.negate(Nx.rsqrt(Nx.subtract(1.0, Nx.power(x, 2.0)))))
```

The `negate` is a bit more deliberate so when you're inspecting a complex expression it's harder to miss what's going on. I think a compiler has a better chance of optimizing `rsqrt` versus the `divide(...)` pattern, but that could be wrong. Either way it's also a bit more deliberate, so harder to miss in debugging :)